library/src/test/java/org/apache/apex/malhar/lib/fs/S3RecordReaderMockTest.java

```diff
@@ -0,0 +1,271 @@
+import com.datatorrent.lib.io.block.ReaderContext;
+import com.datatorrent.lib.io.fs.S3BlockReader;
+
+import static org.mockito.Mockito.mock;
+
+public class S3RecordReaderMockTest
+{
+  private String inputDir;
+  static String outputDir;
+  private static final String FILE_1 = "file1.txt";
+  private static final String FILE_2 = "file2.txt";
+  private static final String FILE_1_DATA = "1234\n567890\nabcde\nfgh\ni\njklmop";
+  private static final String FILE_2_DATA = "qr\nstuvw\nxyz\n";
+
+  public static class TestMeta extends TestWatcher
+  {
+    public String baseDirectory;
+
+    @Override
+    protected void starting(org.junit.runner.Description description)
+    {
+      this.baseDirectory = "target/" + description.getClassName() + "/" + description.getMethodName();
+    }
+
+  }
+
+  @Rule
+  public TestMeta testMeta = new TestMeta();
+
+  @Before
+  public void setup() throws Exception
+  {
+    inputDir = testMeta.baseDirectory + File.separator + "input";
+
+    FileUtils.writeStringToFile(new File(inputDir + File.separator + FILE_1), FILE_1_DATA);
+    FileUtils.writeStringToFile(new File(inputDir + File.separator + FILE_2), FILE_2_DATA);
+  }
+
+  @Test
+  public void testDelimitedRecords() throws Exception
+  {
+
+    DelimitedApplication app = new DelimitedApplication();
+    LocalMode lma = LocalMode.newInstance();
+    Configuration conf = new Configuration(false);
+    conf.set("dt.operator.S3RecordReaderModuleMock.prop.files", inputDir);
+    conf.set("dt.operator.S3RecordReaderModuleMock.prop.blocksThreshold", "1");
+    conf.set("dt.operator.S3RecordReaderModuleMock.prop.scanIntervalMillis", "10000");
+
+    lma.prepareDAG(app, conf);
+    LocalMode.Controller lc = lma.getController();
+    lc.setHeartbeatMonitoringEnabled(true);
+    lc.runAsync();
+
+    Set<String> expectedRecords = new HashSet<String>(Arrays.asList(FILE_1_DATA.split("\n")));
+    expectedRecords.addAll(Arrays.asList(FILE_2_DATA.split("\n")));
+
+    while (DelimitedValidator.records.size() != expectedRecords.size()) {
+      LOG.debug("Waiting for app to finish");
+      Thread.sleep(1000);
+    }
+    lc.shutdown();
+    Assert.assertEquals(expectedRecords, DelimitedValidator.records);
+
+  }
+
+  private static class S3RecordReaderMock extends S3RecordReader
+  {
+    AmazonS3 s3ClientObject;
+
+    @Override
+    protected FSDataInputStream setupStream(FileBlockMetadata block) throws IOException
+    {
+      if (mode == RECORD_READER_MODE.DELIMITED_RECORD) {
+        ((S3DelimitedRecordReaderContextMock)readerContext).setFilePath(block.getFilePath());
+        ((S3DelimitedRecordReaderContextMock)readerContext).setFileLength(block.getFileLength());
+      } else {
+        ((S3FixedWidthRecordReaderContextMock)readerContext).setFilePath(block.getFilePath());
+        ((S3FixedWidthRecordReaderContextMock)readerContext).setFileLength(block.getFileLength());
+      }
+      return fs.open(new Path(block.getFilePath()));
+    }
```

Comment:
Change to :
  super.setupStream(block);
  return fs.open(new Path(block.getFilePath()));